packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -38,6 +39,10 @@ export class Permissions {
     return this.checkProjectFilterPermission(metric, "createMetrics");
   };
 
+  public canAddComment = (projects: string[]): boolean => {
```

Comment:
open to suggestions on how to type the arguments here - we check for this permission across metrics (with have `projects`, along with ideas, experiments, and features, which only have a `project` property. 

It'd be nice to standardize so the engineer doesn't have to have much context when calling `canAddComment`. 

Currently, it still seems like a step in the right direction, as before it could be a string, or undefined, or string[], so this at least simplifies it a bit.

Reply:
One option is to type the argument like this:

```ts
type ObjWithProject = {project?: string; projects?: string[]};
```

That way, you could call it directly with the thing you want to comment on:
```
canAddComment(feature);
canAddComment(metric);
```

Another option is to make separate `canCommentOnFeature`, `canCommentOnMetric`, etc. methods.  They could all check the same `addComments` permission for now, but it leaves the door open for us to make this finer grained in the future.

Reply:
I'm also fine with how you have it